### PR TITLE
Fix debian symbol not found for test_HttpTransact

### LINKED ISSUE #9601: 10.0.x: debian master builds fail `make check` due to `undefined symbol: http_rsb`
The CI branch builds are demonstrating an issue with `make check` runs on latest master. Running some of the tests fail with `undefined symbol: http_rsb`. Here's a recent log:

https://ci.trafficserver.apache.org/job/master/job/os_build/15948/console

```
FAIL: test_HttpTransact
=======================

/home/jenkins/workspace/master/os_build/src/proxy/http/.libs/test_HttpTransact: symbol lookup error: /home/jenkins/workspace/master/os_build/src/proxy/http/.libs/test_HttpTransact: undefined symbol: _Z31forceLinkRegressionHttpTransactv
FAIL test_HttpTransact (exit status: 127)
```

We need to figure out how to address this symbol issue.

### PR DESCRIPTION
**Pull request by lzx404243**
As reported in #9601, running the unit test `test_HttpTransact` result in the  "symbol not found" error on Debian when ATS is configured with the `enable-hardening` flag.

This PR adds a bunch of stubs to resolve these errors. Suggestions on any cleaner way to resolve this is also welcome :)

This resolves #9601.

### CODE REVIEW

**Comment by lzx404243** [comment]
The docker image I am doing the testing on: `ci.trafficserver.apache.org/ats/debian:11`

**Comment by maskit** [comment]
I'm the guy who added the ignore-all flag. I tried to link dependent objects in a normal way, but it was too painful and ended up adding the flag. I'm fine with removing the flag if we can run the tests, but we should probably make CI green first, and have the hardening flag one of the GitHub CI jobs so we won't be surprised after merging PRs.

**Comment by bneradt** [comment]
> and have the hardening flag one of the GitHub CI jobs so we won't be surprised after merging PRs.

+1. Once we do whatever fix we agree too with addressing this, I'll configure the debian PR CI to run with `--enable-hardening`

**Comment by cmcfarlen** [comment]
Using the LDADD flags from the traffic_server makefile does work to link this test.  Could use that, or start with that and try to prune some back.

**Comment by maskit** [comment]
> Using the LDADD flags from the traffic_server makefile does work to link this test.

Well, linking everything would work for all tests for sure, but I don't want to lose the hints (objects we currently link and the stubs) we gathered so far.

@lzx404243 Can we remove the ignore-all flag if we have these stubs? Or does it still does some work?

**Comment by lzx404243** [comment]
@maskit After removing the ignore-all flag, there would be some compile-time errors such as the following:
```
/usr/bin/ld: ../../iocore/eventsystem/libinkevent.a(UnixEventProcessor.o): undefined reference to symbol 'hwloc_bitmap_snprintf'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libhwloc.so.15: error adding symbols: DSO missing from command line
```

**Comment by maskit** [comment]
@lzx404243 Thanks for checking it.

I'd take this as it is, but leave the decision to @cmcfarlen .

**Comment by lzx404243** [comment]
@cmcfarlen Thanks for your input. I removed the stubs and the `ignore-all` flag, and now linking with the libs from traffic_server LDADD (with some pruning). Let me know if you have any further comments or suggestions.

**Review by bneradt — CHANGES_REQUESTED** [review]

**Review by cmcfarlen — APPROVED** [review]
Awesome, thanks!

File: proxy/http/Makefile.am [hunk 1]
```diff
@@ -129,22 +129,31 @@ test_HttpTransact_CPPFLAGS = \
...
 	$(top_builddir)/src/tscore/libtscore.la \
+	$(top_builddir)/src/tscpp/util/libtscpputil.la \
+	$(top_builddir)/proxy/libproxy.a \
+	$(top_builddir)/iocore/net/libinknet.a \
 	$(top_builddir)/src/records/librecords_p.a \
-	$(top_builddir)/proxy/ProxySession.o \
-	$(top_builddir)/proxy/http/HttpTransact.o \
-	$(top_builddir)/proxy/http/HttpTransactHeaders.o \
-	$(top_builddir)/proxy/hdrs/libhdrs.a \
 	$(top_builddir)/iocore/eventsystem/libinkevent.a \
```
[the remark this answers is not in the document]
Comment (bneradt) [reply proxy/http/Makefile.am:154]: Ah, nevermind. Since these are archive files, they'll be order dependent.

**Review by bneradt — APPROVED** [review]

**Comment by zwoop** [comment]
@bneradt This has merge conflicts on 9.2.x, can you please make a 9.2.x branch PR please?

**Pull request merged**